File: packages/python/taiwan_payroll/_identity.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class IdentityRules:
    labor_applies: bool
    pension_applies: bool
    employment_insurance_allowed: bool


IDENTITY_RULES = {
    "category1": IdentityRules(True, True, True),
    "migrantGeneral": IdentityRules(True, False, False),
    "migrantDomestic": IdentityRules(False, False, False),
}


def get_identity_rules(identity: str) -> IdentityRules:
    rules = IDENTITY_RULES.get(identity)
    if rules is None:
        raise ValueError(f'Unsupported identity "{identity}". Supported: {", ".join(IDENTITY_RULES)}.')
    return rules

File: packages/core/src/identity.ts
import type { Identity } from './types';

export interface IdentityRules {
  laborApplies: boolean;
  pensionApplies: boolean;
  employmentInsuranceAllowed: boolean; // false → 就保 never applies (勞保 11.5%); 健保/職災 always apply
}

export const IDENTITY_RULES: Record<Identity, IdentityRules> = {
  category1: { laborApplies: true, pensionApplies: true, employmentInsuranceAllowed: true },
  migrantGeneral: { laborApplies: true, pensionApplies: false, employmentInsuranceAllowed: false },
  migrantDomestic: { laborApplies: false, pensionApplies: false, employmentInsuranceAllowed: false },
};

export function getIdentityRules(identity: Identity): IdentityRules {
  const rules = IDENTITY_RULES[identity];
  if (!rules) {
    throw new Error(`Unsupported identity "${identity}". Supported: ${Object.keys(IDENTITY_RULES).join(', ')}.`);
  }
  return rules;
}
